python/dgl/graph_index.py

```diff
@@ -127,6 +140,7 @@ def number_of_nodes(self):
         int
             The number of nodes
         """
+        assert self._handle is not None
```

Comment:
The assertion is not needed. Or you could think that all APIs require such assertion which means we should assume the _handle is never None after initialization.